packages/editor/src/utils/dom.js

```diff
@@ -25,6 +25,39 @@ export function getBlockFocusableWrapper( clientId ) {
 	return getBlockDOMNode( clientId ).closest( '.editor-block-list__block' );
 }
 
+/**
+ * Returns the expected width of a block which would occupy the editor block
+ * list, in absolute pixels. This value is cached; the cache reset upon change
+ * in viewport size.
+ *
+ * @return {number} Expected block width in pixels.
+ */
+export const getBlockWidth = ( () => {
+	let width;
+	window.addEventListener( 'resize', () => width = undefined );
+
+	return () => {
+		if ( width === undefined ) {
+			const layout = document.querySelector( '.editor-block-list__layout' );
+			if ( ! layout ) {
+				return;
+			}
+
+			const measure = document.createElement( 'div' );
+			measure.className = 'wp-block editor-block-list__block';
+			layout.appendChild( measure );
```

Comment:
To avoid the arbitrary `clientWidth - 30`, perhaps can append another div here with `class="editor-block-list__block-edit"` to mimic the default block layout, then get `clientWidth` from it. I'll try it.

I know, it's fragile, but seems workable for now.